ansible/roles/csm.cn.dns-resolver/tasks/main.yaml

```diff
@@ -0,0 +1,64 @@
+# Copyright 2020-2021 Hewlett Packard Enterprise Development LP
+# resolver config tasks
+---
+- block:
+  - name: Slurp in /etc/resolv.conf to get any DHCP added entries
+    slurp:
+      src: "/etc/resolv.conf"
+    register: resolv_conf
+
+  - name: Register /etc/resolv.conf contents
+    set_fact:
+      resolv: "{{ resolv_conf.content | b64decode }}"
+
+  - name: Parse existing options from /etc/resolv.conf
+    set_fact:
+      node_existing_options: "{{ resolv | regex_findall( 'options (\\b(?:[0-9]{1,3}\\.){3}[0-9]{1,3})\\b') }}"
+
+  - name: Slurp in /etc/sysconfig/network/config to get any DHCP added entries
+    slurp:
+      src: "/etc/sysconfig/network/config"
+    register: net_config
+
+  - name: Register /etc/sysconfig/network/config contents
+    set_fact:
+      netconf: "{{ net_config.content | b64decode }}"
+
+  - name: Parse options from config list
+    set_fact:
+      node_netconf_options: "{{ item.split() }}"
+    loop: "{{ netconf | default() | regex_findall( 'NETCONFIG_DNS_RESOLVER_OPTIONS=\"(.*)\"') }}"
+
+  - name: Add existing options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: (node_existing_options is defined) and (item not in node_netconf_options)
+    loop: "{{ node_existing_options }}"
+
+  - name: Add required options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: item not in node_netconf_options
+    loop: "{{ node_required_dns_options }}"
+
+  - name: Add any customer defined options to node_netconf_options
+    set_fact:
+      node_netconf_options: "{{ node_netconf_options + [ item ] }}"
+    when: item not in node_netconf_options
+    loop: "{{ node_external_dns_options }}"
+
+  - name: Set DNS_RESOLVER_OPTIONS entries
+    when: node_netconf_options is defined
+    lineinfile:
+      dest: /etc/sysconfig/network/config
+      regex: '^NETCONFIG_DNS_RESOLVER_OPTIONS=\"(.*)\"$'
+      line: "NETCONFIG_DNS_RESOLVER_OPTIONS=\"{{ node_netconf_options | join(' ') }}\""
+      backrefs: yes
+
+  - name: Generate updated /etc/resolv.conf by restarting wicked
+    shell: "systemctl restart wicked;sleep 5"
```

Comment:
This might generate a new `/etc/resolv.conf` but it's not deterministic, and if it was used this way it should use the `service` module.

You can deterministically generate a new `/etc/resolv.conf` while removing the 5 second wait and loss of connectivity by doing a:

```yaml
shell: netconfig update -f
```

Could even add a `rm /etc/resolv.conf` or a `mv /etc/resolv.conf /etc/resolv.conf.bak` beforehand.

Reply:
It seems like just using the systemd module to restart the service would be the way to go. I'll go that route.